### PR TITLE
chore: miscellaneous RNTuple improvements

### PR DESCRIPTION
**Pull request by ariostas**
This PR addresses the issues in #1220. I fixed the `__len__` method so that it returns the number of top-level columns instead of the number of entries, which is now the `num_entries` property.

I'll work on implementing an `iterate` method and update this description when it's ready.

### CODE REVIEW

File: src/uproot/const.py [hunk 1]
```diff
@@ -212,7 +208,29 @@
...
+rntuple_field_flag_checksum = 0x04
+
+rntuple_col_flag_deferred = 0x08
+
+rntuple_extra_type_identifier_root = 0x00
+
+rntuple_user_metadata_type_int = 0x01
+rntuple_user_metadata_type_bool = 0x02
+rntuple_user_metadata_type_double = 0x03
+rntuple_user_metadata_type_string = 0x04
```
Comment (ariostas) [inline src/uproot/const.py:236]: Maybe these should be enums

**Comment by ariostas** [comment]
I made some more updates to keep up to date with the spec. I'll wrap up this PR next week and possibly just postpone the implementation of `iterate` to another PR.

Closes #1288, closes #1302, closes #1303

**Comment by ariostas** [comment]
@jpivarski this PR is ready for review now. Here are some highlights/comments:

- I fixed the `__len__` vs `num_entries` issue from #1220.
- I implemented a simple `iterate` method. It supports step sizes in terms of memory, but the current estimation method might not be optimal.
- I added an `RNTupleField` class, so that now you can do things like `obj["nMuon"].array()` and `obj["AntiKt4TruthDressedWZJetsAux:"]["pt"].array()`, but I haven't implemented path support yet for things like `obj["AntiKt4TruthDressedWZJetsAux:/pt"].array()`.
- I updated some things to match the spec, including fixes to #1288 #1302 #1303.
- I made some other minor improvements.

**Review by jpivarski — APPROVED** [review]
Very nice—I like the grouped enums! As far as I can see, these are maintainability updates, not changes in how the code functions. (Okay, I see that the PR title contains "chore:" rather than "feat:" or "fix:".)

Again, I'm forwarding you (email) the information about the RC3 format change, which is likely to be the last one ever.

Feel free to merge this PR!

**Pull request merged**